src/xrpld/app/tx/detail/Escrow.cpp

```diff
@@ -904,7 +903,31 @@ escrowUnlockApplyHelper<Issue>(
         finalAmt = amount.value() - xferFee;
     }
 
-    // If destination is not the issuer then transfer funds
+    // validate the line limit if the acount submitting txn is not the receiver
+    // of the funds
+    if (!createAsset)
+    {
+        auto const sleRippleState = view.peek(trustLineKey);
+        // if the sender is the high, then we use the low limit
+        // otherwise we use the high limit
+        STAmount const lineLimit = sleRippleState->getFieldAmount(
+            senderHigh ? sfLowLimit : sfHighLimit);
+
+        STAmount lineBalance = sleRippleState->getFieldAmount(sfBalance);
+
+        // flip the sign of the line balance if the sender is not high
+        if (!senderHigh)
+            lineBalance.negate();
+
+        // add the final amount to the line balance
+        lineBalance += finalAmt;
+
+        // if the transfer would exceed the line limit return tecPATH_PARTIAL
+        if (lineLimit < lineBalance)
+            return tecPATH_PARTIAL;
```

Comment:
tecPATH_PARTIAL is returned only if payment engine is involved, which isn't the case here. We should return something like `tecLIMIT_EXCEEDED`

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5185/commits/5a4b0d483d1c4ef80816d64bdb16a767f8a10233